compiler/src/iree/compiler/Codegen/LLVMGPU/TransformExtensions/LLVMGPUExtensions.cpp

```diff
@@ -604,14 +606,30 @@ static void populateWarpExecuteOnLane0ToScf(
 
 DiagnosedSilenceableFailure
 transform_dialect::VectorWarpDistributionOp::applyToOne(
-    transform::TransformRewriter &rewriter, Operation *target,
+    transform::TransformRewriter &rewriter, func::FuncOp target,
     transform::ApplyToEachResultList &results,
     transform::TransformState &state) {
-  if (!target->hasTrait<OpTrait::IsIsolatedFromAbove>()) {
-    target->emitOpError(
-        "applies only to isolated-from-above targets because it "
-        "needs to apply "
-        "patterns greedily");
+  if (!isa<HAL::ExecutableOp, HAL::ExecutableVariantOp>(state.getTopLevel())) {
```

Comment:
One of the issues with transform dialect usage is it has a lot of redundant code. For example https://github.com/openxla/iree/blob/85a5425cc8c0b5c4c1db2aa1e242df604bf340ff/compiler/src/iree/compiler/Codegen/Utils/Utils.h#L41 already exists in IREE to do exactly this. Is there a way to reuse?

Reply:
Yeah, these patterns need to be cleaned up. Does the most recent change work for now? There are other places in this file that could use a similar refactor in a follow up.